tests/core/_orchestrator/_dispatcher/test_standalone_job_dispatcher.py

```diff
@@ -0,0 +1,133 @@
+    assert submit_first_call[0].job_id == job.id
+    assert submit_first_call[0].task == task
+    assert submit_first_call[1] == ()
+    assert submit_first_call[2]["config_as_string"] == _TomlSerializer()._serialize(Config._applied_config)
+
+    # test that the proc of the job is added to the list of dispatched jobs
+    assert len(dispatcher.set_dispatch_processes_calls) == 1
+    assert dispatcher.set_dispatch_processes_calls[0][0] == job.id
+    assert dispatcher.set_dispatch_processes_calls[0][1] == dispatcher._executor.f[0]
+
+    # test that the worker is released after the job is done
+    assert len(dispatcher.release_worker_calls) == 1
+
+    # test that the job status is updated after execution on future
+    assert len(dispatcher.update_job_status_from_future_calls) == 1
+    assert dispatcher.update_job_status_from_future_calls[0][0] == job
+    assert dispatcher.update_job_status_from_future_calls[0][1] == dispatcher._executor.f[0]
+
+
+def test_release_worker():
+    dispatcher = _StandaloneJobDispatcher(_OrchestratorFactory._orchestrator)
+
+    assert dispatcher._nb_available_workers == 1
+    dispatcher._release_worker(None)
+    assert dispatcher._nb_available_workers == 2
+    dispatcher._release_worker(None)
+    assert dispatcher._nb_available_workers == 3
+
+
+def test_update_job_status_from_future():
+    task = create_task()
+    job = Job(JobId("job"), task, "s_id", task.id)
+    orchestrator = _OrchestratorFactory._build_orchestrator()
+    dispatcher = _StandaloneJobDispatcher(orchestrator)
+    ft = Future()
+    ft.set_result(None)
+    dispatcher._set_dispatched_processes(job.id, ft)  # the job is dispatched to a process
+
+    dispatcher._update_job_status_from_future(job, ft)
+
+    assert len(dispatcher._dispatched_processes) == 0  # the job process is not stored anymore
+    assert job.is_completed()
+
+
+def test_run():
+    task = create_task()
+    job_1 = Job(JobId("job1"), task, "s_id", task.id)
+    job_2 = Job(JobId("job2"), task, "s_id", task.id)
+    job_3 = Job(JobId("job3"), task, "s_id", task.id)
+    job_4 = Job(JobId("job4"), task, "s_id", task.id)
+    _JobManagerFactory._build_manager()._set(job_1)
+    _JobManagerFactory._build_manager()._set(job_2)
+    _JobManagerFactory._build_manager()._set(job_3)
+    _JobManagerFactory._build_manager()._set(job_4)
+    orchestrator = _OrchestratorFactory._build_orchestrator()
+    orchestrator.jobs_to_run.put(job_1)
+    orchestrator.jobs_to_run.put(job_2)
+    orchestrator.jobs_to_run.put(job_3)
+    orchestrator.jobs_to_run.put(job_4)
+
+    with mock.patch("taipy.core._orchestrator._dispatcher._job_dispatcher._JobDispatcher._execute_job") as mck:
+        dispatcher = _StandaloneJobDispatcher(orchestrator)
+        dispatcher.start()
+        assert_true_after_time(lambda: mck.call_count == 4, msg="The 4 jobs were not dequeued.", time=5)
+        dispatcher.stop()
+        mck.assert_called_with(job_4)
```

Comment:
```suggestion
        mck.assert_called_with(job_1)
        mck.assert_called_with(job_2)
        mck.assert_called_with(job_3)
        mck.assert_called_with(job_4)
```
Can this be done to make sure all jobs are executed?

Reply:
![image](https://github.com/Avaiga/taipy/assets/88036007/f9422310-0be0-47b2-bf3d-67ba443ae9ea)

source: https://docs.python.org/3/library/unittest.mock.html

However I can use `assert_has_calls` instead